cleanco/classify.py

```diff
@@ -3,59 +3,69 @@
 
 Examples of use:
 
->> # check name for its possible business type(s)
->> classification_sources = typesources()
->> matches("MyCompany Ltd", classification_sources)
-['Limited']
->>
-
->> # check name for its possible jurisdictions, usually countries
->> classification_sources = countrysources()
->> matches("MyCompany Ltd", classification_sources)
-['New Zealand', 'United Kingdom', 'United States of America']
->>
+# check name for its possible business type(s)
+>> matches("MyCompany Ltd", "local_name")
+['Private Limited by Guarantee', 'Private Limited Company', 'Limited Liability Company', 'New Generation Co-operatives']
+
+# check name for its possible countries
+>> matches("MyCompany Ltd", "country")
+['United States of America', 'United Kingdom', 'Canada']
 
 """
 
-from .termdata import terms_by_country, terms_by_type
+from iso20275 import Elf, ElfEntry
 from .clean import strip_tail, normalized
 
 
-def typesources():
-   "business types / abbreviations sorted by length of business type"
-   types = []
-   for business_type in terms_by_type:
-       for item in terms_by_type[business_type]:
-           types.append((business_type, item))
+def classifiers():
+    property_names = [p for p in dir(ElfEntry) if isinstance(
+        getattr(ElfEntry, p), property)]
+    return(property_names)
 
-   return sorted(types, key=lambda part: len(part[1]), reverse=True)
 
+def classification(abbreviation, source):
+    classifier_list = []
+    ElfCodeList = []
+    abbreviation = abbreviation
 
-def countrysources():
-   "business countries / type abbreviations sorted by length of type abbreviations"
-   countries = []
-   for country in terms_by_country:
-       for item in terms_by_country[country]:
-           countries.append((country, item))
+    for elf_code, values in Elf.items():
+        entity_codes = Elf[elf_code][0].local_abbreviations
+        if ";" in entity_codes:
+            entity_codes = entity_codes.split(';')
+        if abbreviation in entity_codes:
+            ElfCodeList.append(elf_code)
+        else:
+            pass
 
-   return sorted(countries, key=lambda part: len(part[1]), reverse=True)
+    for item in ElfCodeList:
+        test = getattr(Elf[item][0], source)
+        classifier_list.append(test)
+    # Unique values returned
+    myset = set(classifier_list)
+    classifier_list = list(myset)
+    return(classifier_list)
 
 
-def matches(name, sources):
-    "get types or countries matching with the legal terms in name"
+def matches(name, source):
 
     name = strip_tail(name)
     parts = name.split()
     nparts = [normalized(p) for p in parts]
+    match_list = []
     matches = []
-    for classifier, term in sources:
-        nterm = normalized(term)
+
+    for term in nparts:
         try:
-            idx = nparts.index(nterm)
+            nterm = normalized(term)
+            classification_list = classification(nterm, source)
```

Comment:
Instead of running the classification routine for each part of name, how about generating a structure with all classifiers once in the beginning and just matching name parts against that?